resource/graph_node.go

```diff
@@ -32,14 +50,18 @@ type GraphNode struct {
 	current                   Resource
 	currentModel              Model
 	config                    Config
-	needsReconfigure          bool
 	lastReconfigured          *time.Time
```

Comment:
Would you mind adding a comment above `lastReconfigured` stating its purpose (particularly to distinguish it from `transitionedAt`)? Something like:
```
lastReconfigured stores the timestamp at which the resource was constructed or last reconfigured. 
transitionedAt is generally a more useful timestamp, but `lastReconfigured` is still returned as part
of resource status.
```
`transitionedAt` and the `MachineStatus` endpoint will be a lot more informative than `lastReconfigured` and the `Status` endpoint.